### PR TITLE
User Login for regular user

### PR DESCRIPTION
**Pull request by rer7891**
## What functionality does this accomplish?

**Description:**
Starts functionality of story 13. Can log in regular user but not merchant or admin users.
## What did you struggle on to complete?
Need to add more sad path testing.
### Test Coverage Percentage: 100%
- [ ] No Tests have been changed
- [x] Some Tests have been changed - Adds Login to User Spec
- [ ] All of the Tests have been changed(Please describe what in the world happened):
- [ ] New Tests added for new features

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [x] I have partially tested my code (please explain why): Sad path testing is part of next user story so working on that right now.

## Helpful Resources:
* Resource link AND small description of info gathered/learned
https://backend.turing.io/module2/lessons/authentication

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  helper_method :cart
+  helper_method :cart, :current_user
 
+  def current_user
+    User.find(session[:user_id])
```
Comment (wrenisit) [inline app/controllers/application_controller.rb]: Great job using the sessions!

File: app/views/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<%= form_tag '/login', method: :post do %>
+
+<%= label_tag :email %>
+<%= email_field_tag :email %><br>
+
+<%= label_tag :password %>
+<%= password_field_tag :password %><br>
```
Comment (wrenisit) [inline app/views/sessions/new.html.erb]: Password field is perfect here. Great job.

File: spec/features/users/login_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe "As a visitor when I visit login path", type: :feature do
+  describe "I see a field to enter my email address and password" do
+    it "when I submit valid information I am redirected back to my profile page" do
+
+      regular_user = User.create!(name: "Becky",
```
Comment (wrenisit) [inline spec/features/users/login_spec.rb]: Don't forget to remove your bangs before pushing in the future

File: spec/features/users/login_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      fill_in :password, with: "notsecure123"
+      fill_in :password_confirmation, with: "notsecure123"
+      click_button "Log In"
+
+      expect(current_path).to eq("/profile")
+
+      expect(page).to have_content("Welcome go@foogle.com")
+    end
+
+    xit "when I am a merchant user I am redirected to my merchant dashboard" do
```
Comment (wrenisit) [inline spec/features/users/login_spec.rb]: Thank you for setting placeholders for future use

Comment (jfangonilo) [inline spec/features/users/login_spec.rb]: Looks good, we'll have to add tests here once we get authorization set up

File: spec/features/users/login_spec.rb (hunk 3 above)
Comment (wrenisit) [reply spec/features/users/login_spec.rb]: Maybe not... It's up for debate

**Pull request merged**